src/box/memtx_tx.c

```diff
@@ -1128,8 +1098,100 @@ memtx_tx_story_unlink_top(struct memtx_story *story, uint32_t idx)
 			memtx_tx_ref_to_primary(old_story);
 		memtx_tx_unref_from_primary(story);
 	}
+}
+
+/**
+ * Unlink a @a story from history chain in @a index (in both directions),
+ * where old_top was at the top of chain (that means that index itself
+ * stores a pointer to story->tuple).
+ * This is a light version of function, intended for the case when the
+ * appropriate change in will be done later by caller.
+ */
+static void
+memtx_tx_story_unlink_top_light_common(struct memtx_story *story, uint32_t idx)
+{
+	assert(story != NULL);
+	assert(idx < story->index_count);
+	struct memtx_story_link *link = &story->link[idx];
+	assert(link->newer_story == NULL);
+	struct memtx_story *old_story = link->older_story;
+	if (old_story != NULL)
+		memtx_tx_story_unlink(story, old_story, idx);
+}
+
+/**
+ * See description of `memtx_tx_story_unlink_top_light_common`.
+ * As opposed to `memtx_tx_story_unlink_top_light_on_space_delete`, also
+ * rebinds gap records to the new top in history chain.
+ */
+static void
+memtx_tx_story_unlink_top_light_on_rollback(struct memtx_story *story,
```

Comment:
`memtx_tx_story_unlink_top_light_on_rollback` is a light version of `memtx_tx_story_unlink_top_on_rollback`, so `light` in the end of the method name would be a better variant. The same for all other methods.

Reply:
Yeah, good point, renamed as you suggested.